test/acceptance/CheckoutSpec.scala

```diff
@@ -1,44 +1,89 @@
 package acceptance
 
 import acceptance.pages.{Checkout, ThankYou}
-import org.openqa.selenium.WebDriver
+import acceptance.util.{Util, TestUser, Config, Acceptance}
 import org.scalatest.selenium.WebBrowser
 import org.scalatest.{BeforeAndAfterAll, BeforeAndAfter, FeatureSpec, GivenWhenThen}
+import org.slf4j.LoggerFactory
 
-class CheckoutSpec extends FeatureSpec
-  with WebBrowser with Util with GivenWhenThen with BeforeAndAfter with BeforeAndAfterAll  {
+class CheckoutSpec extends FeatureSpec with WebBrowser with Util
+  with GivenWhenThen with BeforeAndAfter with BeforeAndAfterAll  {
 
-  implicit lazy val driver: WebDriver = Config.driver
+  def logger = LoggerFactory.getLogger(this.getClass)
 
-  // Before each test ...
-  before { resetDriver() }
+  before {
+    resetDriver()
+  } // Before each test
+
+  override def beforeAll() = {
+    Config.printSummary()
+  }
 
   // After all tests execute, close all windows, and exit the driver
-  override def afterAll(): Unit = { quit() }
+  override def afterAll() = {
+    Config.driver.quit()
+  }
 
-  feature("Checkout page") {
-    scenario("Guest user completes a checkout and sets an account password", Acceptance) {
-      val checkout = new pages.Checkout
+  feature("Guest user subscription checkout") {
+    scenario("Guest user subscribes with direct debit", Acceptance) {
+      val checkout = new pages.Checkout(new TestUser)
       When("I visit the checkout page ")
       go.to(checkout)
 
       And("I fill in personal details")
       checkout.fillInPersonalDetails()
 
-      And("I fill in payment details")
-      checkout.fillInPaymentDetails()
+      And("I fill in direct debit payment details")
```

Comment:
As discussed, when you refactor it might be good to look at a way of explicitly linking the test step and the actions performed - e.g. having a map of strings to functions or something similar